app/marketing/views.py

```diff
@@ -604,18 +604,67 @@ def email_settings(request, key):
             else:
                 es.metadata['ip'].append(ip)
             es.save()
-            msg = "Updated your preferences.  "
+            msg = _('Updated your preferences.')
     context = {
         'nav': 'internal',
         'active': '/settings/email',
         'title': _('Email Settings'),
         'es': es,
         'msg': msg,
-        'navs': settings_navs,
+        'navs': get_settings_navs(),
     }
     return TemplateResponse(request, 'settings/email.html', context)
 
 
+def slack_settings(request):
+    """Displays and saves user's slack settings.
+
+    Returns:
+        TemplateResponse: The user's slack settings template response.
+
+    """
+    # setup
+    profile, es, user, is_logged_in = settings_helper_get_auth(request)
+    if not es:
+        login_redirect = redirect('/login/github?next=' + request.get_full_path())
+        return login_redirect
+
+    msg = ''
+
+    if request.POST and request.POST.get('submit'):
+        token = request.POST.get('token', '')
+        repos = request.POST.get('repos').split(',')
+        channel = request.POST.get('channel', '')
+        no_access = [repo_name for repo_name in repos if repo_name != '' and not profile.has_repo(repo_name)]
+        if len(no_access):
+            msg = _('You don\'t have access to {}').format(no_access)
```

Comment:
When possible, try to use literal string interpolation / f-strings since they're significantly faster than `format()`